code/datums/components/bumpattack.dm

```diff
@@ -0,0 +1,69 @@
+/**
+ * Add an auto attack on bump behaviour to item
+ * valid_slot: the inventory slot the item could be held in while still bumpattacking with it
+ * proxy_weapon: the weapon that will gain this behaviour
+ */
+#define COOLDOWN_BUMP_ATTACK "bump_attack"
+/datum/component/bumpattack
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	///inventory slot that the item could be stored while still being able to attack with it
+	var/valid_slots
+	var/active = FALSE
+	var/mob/living/wearer
+	///the item that will gain this behaviour
+	var/obj/item/proxy_weapon
+	///cool down between each hit
+	var/attack_cooldown = CLICK_CD_MELEE
+/datum/component/bumpattack/Initialize(valid_slots, obj/item/proxy_weapon)
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+	src.valid_slots = valid_slots
+	if(proxy_weapon)
+		src.proxy_weapon = proxy_weapon
+
+/datum/component/bumpattack/Destroy(force, silent)
+	return ..()
+
```

Comment:
```suggestion
```
we're just calling parent, is this necessary?

Reply:
true